### ISSUE TITLE
Vector memory exhausted (w0244_s03_pLND_r02.png)

### ISSUE THREAD

**Issue by ex-tag** [CSAFE-ISU/handwriter#128]
**Describe the bug**
```R
Processing document w0244_s03_pLND_r02.png...
path in readPNGBinary: /main_dir/data/template_docs/w0244_s03_pLND_r02.png
The following documents could not be processed: w0244_s03_pLND_r02.png
Loading processed template training documents...
Error: vector memory exhausted (limit reached?)
```

**To Reproduce**

1. CSAFE-ISU offers a database of handwriting images:
https://data.csafe.iastate.edu/HandwritingDatabase/#
(Click on "Download All Handwriting Images")
2. This image database can be used to train the CSAFE-ISU Handwriter:
a. Download the zipped image files.
b. Unzip the image files.
c. Copy the image file "w0244_s03_pLND_r02.png" to the folder "main_dir/data/template_docs".
d. When in R, train the CSAFE-ISU Handwriter from the image templates
```R
library('handwriter');

template <- make_clustering_templates(
     template_dir = "/main_dir",
     template_images_dir = "/main_dir/data/template_docs",
     writer_indices = c(7,10),
     max_edges = 25,
     centers_seed = 100,
     graphs_seed = 101,
     K = 40,
     num_dist_cores = 4,
     max_iters = 25);
```

**Expected behavior**
For the image file to be processed.

**Desktop (please complete the following information):**
```R
R version 4.3.3 (2024-02-29)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Los_Angeles
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.3.3
```

**Comment by stephaniereinders** [comment]
Greetings! I have looked into the problem. I downloaded w0244_s03_pLND_r02.png from the CSAFE Handwriting Database and tried running the following code:

```
library(handwriter)

input_dir <- '/Volumes/T7_Shield/CSAFE/handwriting_datasets/CSAFE_docs_test'
output_dir <- '/Volumes/T7_Shield/CSAFE/handwriting_datasets/CSAFE_graphs_testing2'

process_batch_dir(input_dir, output_dir)
```

This produced the following error:
> error with document w0244_s03_pLND_r02.png Error in png::readPNG(path): file is not in PNG format
NULL

Finder on my Mac says that the document is a PNG file and the file extension is correct, so I'm not sure why the png package flags it as not being in PNG format. However, I opened the image in Photoshop and resaved it as a PNG image. Then handwriter was able to process the image without errors. 

I have requested that the original handwriting sample be re-scanned, saved and uploaded to the CSAFE Handwriting Database. We will see if this fixes the problem.
